app/src/main/java/com/devshish/internship/presentation/ui/lyrics/LyricsViewModel.kt

```diff
@@ -42,19 +46,27 @@ class LyricsViewModel(
                 Timber.e(e)
             } finally {
                 _isProgressLoading.value = false
+                isSongStored()
             }
         }
     }
 
-    fun onLikeButtonClick() {
+    fun onStoreSongClick() {
         viewModelScope.launch {
             try {
                 _lyrics.value?.let { repository.storeSong(searchSong, it) }
             } catch (e: Exception) {
                 Timber.e(e)
             } finally {
                 _isLyricsSaved.value = Event(Unit)
+                isSongStored()
             }
         }
     }
+
+    private fun isSongStored() {
+        viewModelScope.launch {
+            _isSongStored.value = repository.isSongStored(searchSong)
+        }
+    }
```

Comment:
Not needed anymore
+ do not run viewmodelScope.launch inside of other context without necessity